packages/labs/context/src/lib/controllers/context-consumer.ts

```diff
@@ -48,41 +51,41 @@ export class ContextConsumer<
 
   private dispatchRequest() {
     this.host.dispatchEvent(
-      new ContextRequestEvent(
-        this.context,
-        (value, unsubscribe) => {
-          // some providers will pass an unsubscribe function indicating they may provide future values
-          if (this.unsubscribe) {
-            // if the unsubscribe function changes this implies we have changed provider
-            if (this.unsubscribe !== unsubscribe) {
-              // cleanup the old provider
-              this.provided = false;
-              this.unsubscribe();
-            }
-            // if we don't support subscription, immediately unsubscribe
-            if (!this.subscribe) {
-              this.unsubscribe();
-            }
-          }
+      new ContextRequestEvent(this.context, this._callback, this.subscribe)
+    );
+  }
+
+  // This function must have stable identity to properly dedupe in ContextRoot
```

Comment:
The entire change in this file is moving the context-request event callback to an instance field so that it can be used for the callback identity check in ContextRoot. The callback body is the same.